### ISSUE TITLE
staticcheck: unhandled type *ir.ArrayConst

### ISSUE THREAD

**Issue by itsdevbear** [golangci/golangci-lint#4662]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've read the typecheck section of the FAQ (https://golangci-lint.run/welcome/faq/#why-do-you-have-typecheck-errors).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

```
ERRO [runner] Panic: nilness: package "ssz" (isInitialPkg: false, needAnalyzeSource: true): internal error: unhandled type *ir.ArrayConst: goroutine 22646 [running]:
runtime/debug.Stack()
        /opt/homebrew/Cellar/go/1.22.2/libexec/src/runtime/debug/stack.go:24 +0x64
github.com/golangci/golangci-lint/pkg/goanalysis.(*action).analyzeSafe.func1()
        /Users/dev/go/pkg/mod/github.com/golangci/golangci-lint@v1.57.2/pkg/goanalysis/runner_action.go:108 +0x21c
panic({0x103149ac0?, 0x1400977fb80?})
        /opt/homebrew/Cellar/go/1.22.2/libexec/src/runtime/panic.go:770 +0x124
honnef.co/go/tools/analysis/facts/nilness.impl.func1({0x14b4057e0, 0x14003600120})
        /Users/dev/go/pkg/mod/honnef.co/go/tools@v0.4.7/analysis/facts/nilness/nilness.go:239 +0x920
honnef.co/go/tools/analysis/facts/nilness.impl.func1({0x103388ff8, 0x14007e22100})
        /Users/dev/go/pkg/mod/honnef.co/go/tools@v0.4.7/analysis/facts/nilness/nilness.go:147 +0x958
honnef.co/go/tools/analysis/facts/nilness.impl(0x14001f7c750, 0x140042db900, 0x1400442ec78)
        /Users/dev/go/pkg/mod/honnef.co/go/tools@v0.4.7/analysis/facts/nilness/nilness.go:246 +0x2c8
honnef.co/go/tools/analysis/facts/nilness.run(0x14001f7c750)
        /Users/dev/go/pkg/mod/honnef.co/go/tools@v0.4.7/analysis/facts/nilness/nilness.go:66 +0xf4
github.com/golangci/golangci-lint/pkg/goanalysis.(*action).analyze(0x14003eaa2c0)
        /Users/dev/go/pkg/mod/github.com/golangci/golangci-lint@v1.57.2/pkg/goanalysis/runner_action.go:190 +0x8c4
github.com/golangci/golangci-lint/pkg/goanalysis.(*action).analyzeSafe.func2()
        /Users/dev/go/pkg/mod/github.com/golangci/golangci-lint@v1.57.2/pkg/goanalysis/runner_action.go:112 +0x20
github.com/golangci/golangci-lint/pkg/timeutils.(*Stopwatch).TrackStage(0x140006131d0, {0x102daeee3, 0x7}, 0x14002c78f30)
        /Users/dev/go/pkg/mod/github.com/golangci/golangci-lint@v1.57.2/pkg/timeutils/stopwatch.go:111 +0x44
github.com/golangci/golangci-lint/pkg/goanalysis.(*action).analyzeSafe(0x10274c760?)
        /Users/dev/go/pkg/mod/github.com/golangci/golangci-lint@v1.57.2/pkg/goanalysis/runner_action.go:111 +0x78
github.com/golangci/golangci-lint/pkg/goanalysis.(*loadingPackage).analyze.func2(0x14003eaa2c0)
        /Users/dev/go/pkg/mod/github.com/golangci/golangci-lint@v1.57.2/pkg/goanalysis/runner_loadingpackage.go:80 +0xb0
created by github.com/golangci/golangci-lint/pkg/goanalysis.(*loadingPackage).analyze in goroutine 2145
        /Users/dev/go/pkg/mod/github.com/golangci/golangci-lint@v1.57.2/pkg/goanalysis/runner_loadingpackage.go:75 +0x184 
```

### Version of golangci-lint

<details>

```console
go run github.com/golangci/golangci-lint/cmd/golangci-lint --version
golangci-lint has version v1.57.2 built with go1.22.2 from (unknown, modified: ?, mod sum: "h1:NNhxfZyL5He1WWDrIvl1a4n5bvWZBcgAqBwlJAAgLTw=") on (unknown)
dev@Devs-MacBook-Pro-2 beacon-kit % 
```

</details>

### Configuration

<details>

```console
# paste configuration file or CLI flags here
```

</details>

### Go environment

<details>

```console
$ go version && go env
# paste output here
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// add your code here
```

</details>

### Validation

- [X] Yes, I've included all information above (version, config, etc.).

**Comment by ldez** [comment]
Hello,

can you provide a reproducible example?

**Comment by ldez** [comment]
It's a problem with Staticcheck https://github.com/dominikh/go-tools/issues/1397

The problem has been fixed by https://github.com/dominikh/go-tools/commit/b3ee554011ba07e55a74da8c2e46de2accf60ec1

But this commit is missing from the latest release branch of staticcheck.

**Comment by itsdevbear** [comment]
Oh wow thanks for digging in! Is there a way to force the usage of latest main for static check within golangci?

**Comment by ldez** [comment]
staticcheck uses release branches, the difference between the main branch and a release branch, is complex to verify.
So I think it is not a good idea to use the main branch.

Linters are classic Go modules, so to use another version of a dependency you have to update the `go.mod` and compile golangci-lint.

---

_Sponsoring is a good way to sustain open source maintainers: [sponsor me](https://github.com/sponsors/ldez)_

**Comment by keyno63** [comment]
Has this issue already been resolved? 

https://github.com/dominikh/go-tools/commit/b3ee554011ba07e55a74da8c2e46de2accf60ec1 is released [go-tools/v0.5.0](https://github.com/dominikh/go-tools/releases/tag/2024.1)
and included in [golangci-lint/v1.60.1](https://github.com/golangci/golangci-lint/releases/tag/v1.60.1)


If there's anything I can assist with, I'd be happy to help

**Comment by ldez** [comment]
My issue is still open: https://github.com/dominikh/go-tools/issues/1522
